README.md

```diff
@@ -78,6 +78,89 @@ export default class MoreFriendz extends Component {
 }
 ```
 
+It can also be used in decorator form to wrap an async function so calls to it are registered with the test waiter system.
+
+```js
+import Component from '@ember/component';
+import { waitForPromise } from 'ember-test-waiters';
+
+export default class MoreFriendz extends Component {
+  @waitForPromise
+  async doAsyncStuff() {
+    await someAsyncWork();
+  }
+
+  didInsertElement() {
+    this.doAsyncStuff().then(() => {
+      doOtherThings();
+    });
+  }
+}
+```
+
+or in non-decorator form to wrap an async function.
+
+```js
+import Component from '@ember/component';
+import { waitForPromise } from 'ember-test-waiters';
+
+export default Component.extend({
+  doAsyncStuff: waitForPromise(async function doAsyncStuff() {
+    await someAsyncWork();
+  }),
+
+  didInsertElement() {
+    this.doAsyncStuff().then(() => {
+      doOtherThings();
+    });
+  }
+})
+```
+
+### waitForCoroutine function
+
+This addon also provides a `waitForCoroutine` function, which can be used to wrap a coroutine, such as an `ember-concurrency` task function, so calls to it are registered with the test waiter system.
```

Comment:
Can you link out to a definition of "coroutine" here?

Reply:
I think we also have to be a bit more specific about what it means to wait for the coroutine. 

Does it wait for completion? Does it wait for the first yield? Something else entirely?

Reply:
I also had to look this up! I mean I _knew_ what a coroutine was in terms of the definition, but I had to read the words to overlay the mental picture onto what this code is doing. I think a more robust description would be ideal.